googleapiclient/discovery.py

```diff
@@ -712,7 +707,7 @@ def method(self, **kwargs):
         raise TypeError('media_filename must be str or MediaUpload.')
 
       # Check the maxSize
-      if maxSize > 0 and media_upload.size() > maxSize:
+      if media_upload.size() and media_upload.size() > maxSize > 0:
```

Comment:
Thinking about it a bit more, I'm thinking that it's better to check for None explicitly instead of just an object that evaluates to False.

Here is the doc for `.size()`

    Returns:
      Size of the body, or None of the size is unknown.

Added commit 8b8af28 for this.